From ed62e9f5246e939d1801b77c3a2b91d16812af11 Mon Sep 17 00:00:00 2001
From: lairuilang <lawrenlang1@gmail.com>
Date: Fri, 24 Apr 2026 14:40:57 +0800
Subject: [PATCH 32/48] urma:fix wrid=0 in empty wr

(cherry picked from commit dfa3174c958e002fb9aacb7dcd5af7abc96d4a88)
---
 src/urma/lib/urma/bond/bondp_wr_buf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/urma/lib/urma/bond/bondp_wr_buf.h b/src/urma/lib/urma/bond/bondp_wr_buf.h
index ddbf9eda..0a08b3dc 100644
--- a/src/urma/lib/urma/bond/bondp_wr_buf.h
+++ b/src/urma/lib/urma/bond/bondp_wr_buf.h
@@ -86,7 +86,7 @@ static inline jfs_wr_entry_t *jfs_wr_buf_get(wr_buf_t *buf, uint64_t wr_id)
 {
     jfs_wr_entry_t *wr_entry;
     wr_entry = (jfs_wr_entry_t *)__wr_buf_idx(buf, __wr_id_to_idx(wr_id, buf->max_wr_num));
-    return wr_entry->wr_id == wr_id ? wr_entry : NULL;
+    return wr_entry->wr_id == wr_id && wr_id != 0 ? wr_entry : NULL;
 }
 
 static inline jfr_wr_entry_t *jfr_wr_buf_get(wr_buf_t *buf, uint64_t wr_id)
-- 
2.28.0.windows.1

